app/src/main/java/com/onboarding/calculatorkotlin/mvp/model/CalculatorModel.kt

```diff
@@ -0,0 +1,60 @@
+package com.onboarding.calculatorkotlin.mvp.model
+
+import com.onboarding.calculatorkotlin.mvp.CalculatorContract
+
+class CalculatorModel : CalculatorContract.Model {
+
+    private var firstOperand = EMPTY_STRING
+    private var secondOperand = EMPTY_STRING
+    private var operator = EMPTY_STRING
+
+    override fun setValue(value: String) {
+        if (operator.isEmpty()) {
+            firstOperand += value
+        } else {
+            secondOperand += value
+        }
+    }
+
+    override fun setOperator(operator: String) {
+        this.operator = operator
+    }
+
+    override fun cleanValue() {
+        if (secondOperand.isNotEmpty()) {
+            secondOperand = secondOperand.dropLast(ONE_VALUE_REMOVED)
+        } else if (operator.isNotEmpty()) {
+            operator = EMPTY_STRING
+        }
+        firstOperand = firstOperand.dropLast(ONE_VALUE_REMOVED)
+    }
+
+    override fun addMinus() {
+        if ((firstOperand.isEmpty() && operator.isEmpty()) || (secondOperand.isEmpty() && operator.isNotEmpty())) {
+            setValue(MINUS)
+        } else {
+            setOperator(MINUS)
+        }
+    }
+
+    override fun cleanAll() {
+        firstOperand = EMPTY_STRING
+        secondOperand = EMPTY_STRING
+        operator = EMPTY_STRING
+    }
+
+    override fun getLastModified(): String {
+        if (secondOperand.isNotEmpty()) {
```

Comment:
Here you can use a return if so avoid 3 times saying return

Reply:
Solved!